crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -2693,6 +2725,9 @@ impl AuthorityPerEpochStore {
                     details: PendingCheckpointInfo {
                         timestamp_ms: commit_timestamp,
                         last_of_epoch: final_round,
+                        // Note if randomness is not generated for this commit, we will skip the
+                        // checkpoint with the associated height. Therefore checkpoint heights may
+                        // not be contiguous.
```

Comment:
isn't randomness already known to be generated at this point? do you mean "if randomnes is not generated for some commit"?

Reply:
Yes, it is known as of line 2710, I meant "for some commit" but will move comment up to make it clearer.